src/models/tools/tool-types.ts

```diff
@@ -49,15 +49,27 @@ export const ToolMetadataModel = types.model("ToolMetadataModel", {
   });
 export interface ToolMetadataModelType extends Instance<typeof ToolMetadataModel> {}
 
-export const ToolButtonModel = types.model("ToolButton", {
+const BaseToolButtonModel = types.model("BaseToolButton", {
   name: types.string,
   title: types.string,
-  iconId: types.string,
   isDefault: false,
-  isTileTool: false
 });
-export interface ToolButtonModelType extends Instance<typeof ToolButtonModel> {}
-export interface ToolButtonSnapshot extends SnapshotOut<typeof ToolButtonModel> {}
+
+const AppToolButtonModel = BaseToolButtonModel.named("AppToolButtonModel")
+  .props({
+    iconId: types.string,
+    isTileTool: types.literal(false)
+  });
+
+const TileToolButtonModel = BaseToolButtonModel.named("TileToolButtonModel")
+  .props({
+    isTileTool: types.literal(true)
+  });
+export const ToolButtonModel = types.union(AppToolButtonModel, TileToolButtonModel);
```

Comment:
This approach wasn't totally necessary, but it allows us to enforce that there is no iconId for tool buttons that are tool tiles.  The `isTileTool` is used by MST to figure out which type in the union to use. 
And the generated typescript types correctly handle this so you can't create a ToolButtonModel that is both a isTileTool and has iconId.